File: backend/Shared/EventPlanR.Domain/Repositories/Models/EventFilter.cs
using EventPlanr.Domain.Common;
using EventPlanr.Domain.Enums;

namespace EventPlanr.Domain.Repositories.Models;

public class EventFilter
{
    public string? SearchTerm { get; set; }
    public EventCategory? Category { get; set; }
    public Language? Language { get; set; }
    public Currency? Currency { get; set; }
    public DateTimeOffset? FromDate { get; set; }
    public DateTimeOffset? ToDate { get; set; }
    public LocationFilter? Location { get; set; } = null!;
}

File: frontend/lib/domain/models/event/event_filter.dart
import 'package:event_planr_app/domain/models/common/order_direction_enum.dart';
import 'package:event_planr_app/domain/models/event/currency_enum.dart';
import 'package:event_planr_app/domain/models/event/event_category_enum.dart';
import 'package:event_planr_app/domain/models/event/event_distance_enum.dart';
import 'package:event_planr_app/domain/models/event/event_order_by_enum.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'event_filter.freezed.dart';
part 'event_filter.g.dart';

@freezed
class EventFilter with _$EventFilter {
  const factory EventFilter({
    required OrderDirectionEnum orderDirection,
    required EventOrderByEnum orderBy,
    String? searchTerm,
    EventCategoryEnum? category,
    CurrencyEnum? currency,
    DateTime? fromDate,
    DateTime? toDate,
    String? locationName,
    double? longitude,
    double? latitude,
    EventDistanceEnum? distance,
    int? pageNumber,
    int? pageSize,
  }) = _EventFilter;

  factory EventFilter.fromJson(Map<String, dynamic> json) =>
      _$EventFilterFromJson(json);
}
